internal/zinc-apiinfo/src/main/scala/sbt/internal/inc/ClassToAPI.scala

```diff
@@ -183,9 +183,28 @@ object ClassToAPI {
     val fields = mergeMap(c, c.getDeclaredFields, c.getFields, fieldToDef(c, cf, enclPkg))
     val constructors =
       mergeMap(c, c.getDeclaredConstructors, c.getConstructors, constructorToDef(enclPkg))
+    val cl = c.getClassLoader
+    val name = c.getCanonicalName
```

Comment:
Is the original implementation of `getDeclaredClasses` and `getClasses` method using `getCanonicalName`?

Reply:
I can't read the original implementation to avoid license violation. Here the name has to match what's stored in the classfile, and then later converted to dot-notation via `getClassConstantName`. In any case, I am now thinking `c.getName` might actually be the correct answer since at the classfile level they'll likely store `foo/bar/A$B`.

Reply:
> I can't read the original implementation to avoid license violation. 
My understanding was that this only applies if you're doing a a clean-room implementation of the Java specification, and even then it's not clear what license you'd be violating.

At any rate, OpenJDK is GPL (at least the parts related to java.lang.Class), for instance [here](https://github.com/unofficial-openjdk/openjdk/blob/72b82da47d66e40f5f3f9a5df3e3559a182562e7/src/hotspot/share/prims/jvm.cpp) or [here](https://github.com/dmlloyd/openjdk/blob/jdk/jdk/src/java.base/share/classes/java/lang/Class.java).